From 3ccfd56f3c7f4beb3ad1de69baa7b08b332c5576 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 25 Jul 2014 16:44:53 -0700
Subject: [PATCH 0415/1566] drm: xilinx: plane: Use IS_ERR() for
 of_dma_request_slave_channel()

of_dma_request_slave_channel() returns an error pointer in case of
error, so use IS_ERR() and PTR_ERR() to check and return the correct
error code.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_plane.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
index bbaf9d5..60afe1f 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
@@ -627,9 +627,9 @@ xilinx_drm_plane_create(struct xilinx_drm_plane_manager *manager,
 	DRM_DEBUG_KMS("plane->id: %d\n", plane->id);
 
 	plane->vdma.chan = of_dma_request_slave_channel(plane_node, "vdma");
-	if (!plane->vdma.chan) {
+	if (IS_ERR(plane->vdma.chan)) {
 		DRM_ERROR("failed to request dma channel\n");
-		ret = -ENODEV;
+		ret = PTR_ERR(plane->vdma.chan);
 		goto err_out;
 	}
 
-- 
1.7.5.4

